base/abstractarray.jl

```diff
@@ -1898,7 +1902,8 @@ Concatenate arrays or numbers vertically. Equivalent to [`cat`](@ref)`(A...; dim
 and to the syntax `[a; b; c]`.
 
 To concatenate a large vector of arrays, `reduce(vcat, A)` calls an efficient method
-when `A isa AbstractVector{<:AbstractVecOrMat}`, rather than working pairwise.
+when `A isa AbstractVector{<:AbstractVecOrMat}`, or a tuple of `AbstractVecOrMat`.
+On all other types, `reduce(vcat, A)` works one at a time, making `length(A)-1` vectors.
```

Comment:
```suggestion
On all other types, `reduce(vcat, A)` works one at a time, producing `length(A)-1`
intermediate results.
```

Also, question: couldn't the intermediate results also not be vectors? E.g. they could perhaps be matrices as well, couldn't they?

Reply:
true. Maybe it should say "arrays". The main point is that these intermediates typically involve allocation.

Reply:
```suggestion
On all other types, `reduce(vcat, A)` works one at a time, creating `length(A)-1` arrays, and discarding all but the last.
This less efficient behaviour is also triggered by keywords like `reduce(vcat, A; init=Int[])`.
```